tests/helpers/test_storage_remove.py

```diff
@@ -0,0 +1,35 @@
+"""Tests for the storage helper with minimal mocking."""
+import asyncio
+from datetime import timedelta
+import os
+
+from homeassistant.helpers import storage
+from homeassistant.util import dt
+
+from tests.async_mock import patch
+from tests.common import async_fire_time_changed, async_test_home_assistant
+
+
+async def test_removing_while_delay_in_progress(tmpdir):
+    """Test removing while delay in progress."""
+
+    loop = asyncio.get_event_loop()
+    hass = await async_test_home_assistant(loop)
```

Comment:
Should we make sure we shut this down?

Reply:
Like this https://github.com/home-assistant/core/blob/dev/tests/conftest.py#L127-L133

Reply:
Will do 👍